### PR TITLE
Cleanup amp-form-custom-validations experiment

### LINKED ISSUE #6377: Remove amp-form-var-sub and amp-form-var-sub-for-post experiment
Cleanup issue for 
- [ ] amp-form-var-sub 
- [ ] amp-form-var-sub-for-post

### PR DESCRIPTION
**Pull request by mkhatib**
Closes #6377

### CODE REVIEW

File: build-system/global-configs/canary-config.json [hunk 1]
```diff
@@ -8,7 +8,6 @@
   "expDoubleclickA4A": 1,
   "pan-y": 1,
   "a4aProfilingRate": 1,
-  "amp-form-custom-validations": 1,
```
Comment (zhouyx) [inline build-system/global-configs/canary-config.json]: I believe this should be removed afterwards.  the actual experiment check can only make to canary this week, while the config will be live in prod.

File: build-system/global-configs/prod-config.json [hunk 2]
```diff
@@ -8,7 +8,6 @@
   "expAdsenseA4A": 0.1,
   "expDoubleclickA4A": 0.1,
   "a4aProfilingRate": 1,
-  "amp-form-custom-validations": 1,
```
Comment (zhouyx) [inline build-system/global-configs/prod-config.json]: same here

File: build-system/global-configs/canary-config.json (hunk 1 above)
Comment (mkhatib) [inline build-system/global-configs/canary-config.json]: Ah. I think you're right. @erwinmombay can confirm?

Comment (mkhatib) [inline build-system/global-configs/canary-config.json]: Done

File: build-system/global-configs/prod-config.json (hunk 2 above)
Comment (mkhatib) [inline build-system/global-configs/prod-config.json]: Done

File: build-system/global-configs/canary-config.json (hunk 1 above)
Comment (erwinmombay) [inline build-system/global-configs/canary-config.json]: yes, will need to wait until the guard removal hits prod to not unlaunch the feature

**Pull request merged**